src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs

```diff
@@ -181,9 +165,28 @@ private void Pump()
             finally
             {
                 // If an item throws, we want to ensure that the next pump gets scheduled appropriately regardless.
-                if(!isDisposed) AwaitNewData();
+                if (!_isDisposed) AwaitNewData();
+            }
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (!_isDisposed)
+            {
+                if (disposing)
+                {
+                    Queue.Writer.Complete();
+                }
+                SetSynchronizationContext(previousSynchronizationContext);
```

Comment:
this is incorrect if someone changed it out from under us, and also incorrect if we get disposed from the wrong thread